src/_01_IntroToArrayLists/_06_IPodShuffle.java

```diff
@@ -1,27 +1,68 @@
 package _01_IntroToArrayLists;
 
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
 import java.util.ArrayList;
 import java.util.Random;
 
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
 
 //Copyright The League of Amazing Programmers, 2015
 
-public class _06_IPodShuffle{
+public class _06_IPodShuffle implements ActionListener {
+	JFrame frame = new JFrame();
+	JPanel panel = new JPanel();
+	JButton playButton = new JButton();
+
+	ArrayList<Song> songsList = new ArrayList<>();
+
+	Random rand = new Random();
+	int randomNum;
+	int currentSong;
+
 	public _06_IPodShuffle() {
-		// 1. Use the Song class the play the demo.mp3 file.
-				
-				
-		/**
-		 * 2. Congratulations on completing the sound check! * Now we want to make an
-		 * iPod Shuffle that plays random music. * Create an ArrayList of Songs and a
-		 * "Surprise Me!" button that will play a random song when it is clicked. * If
-		 * you're really cool, you can stop all the songs, before playing a new one on
-		 * subsequent button clicks.
-		 */
-		
+		songsList.add(new Song("ipod-dramatic.mp3"));
+		songsList.add(new Song("ipod-sad.mp3"));
+		songsList.add(new Song("ipod-upbeat.mp3"));
+		songsList.add(new Song("ipod-yeehaw.mp3"));
+	}
+
+	public void actionPerformed(ActionEvent e) {
+		JButton buttonPressed = (JButton) e.getSource();
+
+		if (buttonPressed == playButton && playButton.getText().equals("PLAY")) {
+			setNewRandomNum();
+			songsList.get(randomNum).play();
+			playButton.setText("STOP");
+		} else if (buttonPressed == playButton && playButton.getText().equals("STOP")) {
+			songsList.get(randomNum).stop();
+			playButton.setText("PLAY");
```

Comment:
When you're using the same string value in multiple places, you can save it in a final variable.  By having it defined in just one place, it helps avoid inconsistency problems.  For example, what would happen if you decided you wanted to change "PLAY" to "play" but missed one of the three places where you need to change it?